From 2991beeb10d06fdf2be3a6852dda272c89e8bc00 Mon Sep 17 00:00:00 2001
From: Aakash Sasidharan <Aakash.Sasidharan@cavium.com>
Date: Fri, 19 Oct 2018 10:07:24 +0300
Subject: [PATCH 0338/1051] octeontx: add new fields to mbox_bgx_port_status_t

- Add speed, duplex fields to mbox_bgx_port_status_t structure.
- As modification to existing mbox structures requires updation of
  the MBOX interface version number, increment MBOX interface minor
  version number to 0x1.

Signed-off-by: Aakash Sasidharan <Aakash.Sasidharan@cavium.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/bgx.c           | 8 +++++++-
 drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h | 6 +++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c b/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c
index 478a7997eae0..bd86e1bff691 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/bgx.c
@@ -529,14 +529,20 @@ int bgx_port_config(struct octtx_bgx_port *port, mbox_bgx_port_conf_t *conf)
 int bgx_port_status(struct octtx_bgx_port *port, mbox_bgx_port_status_t *stat)
 {
 	struct bgxpf *bgx;
+	struct bgx_link_status link;
 	u64 reg;
 
 	bgx = get_bgx_dev(port->node, port->bgx);
 	if (!bgx)
 		return -EINVAL;
+
 	reg = bgx_reg_read(bgx, port->lmac, BGX_CMR_RX_BP_STATUS);
 	stat->bp = reg & 0x1; /* BP */
-	stat->link_up = bgx_get_link_status(port->node, port->bgx, port->lmac);
+
+	thbgx->get_link_status(port->node, port->bgx, port->lmac, &link);
+	stat->link_up = link.link_up;
+	stat->duplex = link.duplex;
+	stat->speed = link.speed;
 	return 0;
 }
 
diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h
index 36248244d99c..bf71c46b5194 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h
@@ -268,7 +268,7 @@ struct mbox_intf_ver {
 static const struct mbox_intf_ver MBOX_INTERFACE_VERSION = {
 	.platform = 0x01,
 	.major = 0x00,
-	.minor = 0x00
+	.minor = 0x01
 };
 
 /* FIXME: This union is temporary until we agree to move all messages to RAM */
@@ -352,6 +352,10 @@ typedef struct mbox_bgx_port_status {
 	u8 link_up;
 	/* 1 = LMAC is backpressured, 0 = no backpressure. */
 	u8 bp;
+	/* Duplex mode: 1 = full duplex, 0 = half duplex */
+	u8 duplex;
+	/* Link speed in Mbps */
+	u32 speed;
 } mbox_bgx_port_status_t;
 
 /* BGX port statistics: */
-- 
2.17.1

